elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java

```diff
@@ -96,8 +114,197 @@ private ModelNode assertFailed(ModelNode response) {
+        SelfSignedX509CertificateAndSigningKey anotherRootSelfSignedX509CertificateAndSigningKey = createCertRoot(ANOTHER_ROOT_DN);
+        SelfSignedX509CertificateAndSigningKey sallySelfSignedX509CertificateAndSigningKey = createSallyCertificate();
+
+        PublicKey publicKey = sallySelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate().getPublicKey();
+
+        // Key Stores
+        File fireflyFile = new File(workingDir, "firefly.keystore");
+        File testFile = new File(workingDir, "test.keystore");
+
+        KeyStore fireflyKeyStore = createFireflyKeyStore();
+        KeyStore testKeyStore = createTestKeyStore(rootSelfSignedX509CertificateAndSigningKey, sallySelfSignedX509CertificateAndSigningKey);
+
+        createTemporaryKeyStoreFile(fireflyKeyStore, fireflyFile);
+        createTemporaryKeyStoreFile(testKeyStore, testFile);
+
+        // Cert Files
+        File testSingleCertReplyFile = new File(workingDir, "test-single-cert-reply.cert");
+        File testCertChainReplyFile = new File(workingDir, "test-cert-chain-reply.cert");
+        File testExportedFile = new File(workingDir, "test-exported.cert");
+        File testTrustedFile = new File(workingDir, "test-trusted.cert");
+        File testUntrustedFile = new File(workingDir, "test-untrusted.cert");
+        File testUntrustedCertChainReplyFile = new File(workingDir, "test-untrusted-cert-chain-reply.cert");
+
+        X509Certificate sSmithCertificate = createSSmithCertificate(rootSelfSignedX509CertificateAndSigningKey, publicKey);
+
+        X509Certificate testTrustedCert = createTestTrustedCertificate(rootSelfSignedX509CertificateAndSigningKey);
+        X509Certificate testUntrustedCert = anotherRootSelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate();
+        X509Certificate testUntrustedCertChainReplyCert = createTestUntrustedCertChainReplyCertificate(anotherRootSelfSignedX509CertificateAndSigningKey, publicKey);
+
+        createTemporaryCertFile(sSmithCertificate, testSingleCertReplyFile);
+
+        createTemporaryCertFile(rootSelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate(), testCertChainReplyFile);
+        createTemporaryCertFile(sSmithCertificate, testCertChainReplyFile);
+
+        createTemporaryCertFile(sallySelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate(), testExportedFile);
+        createTemporaryCertFile(testTrustedCert, testTrustedFile);
+        createTemporaryCertFile(testUntrustedCert, testUntrustedFile);
+
+        createTemporaryCertFile(testUntrustedCertChainReplyCert, testUntrustedCertChainReplyFile);
+        createTemporaryCertFile(anotherRootSelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate(), testUntrustedCertChainReplyFile);
+    }
+
+    private static void removeTestFiles() {
+        File workingDir = new File(WORKING_DIRECTORY_LOCATION);
+        if (workingDir.exists() == false) {
+            workingDir.mkdirs();
```

Comment:
No need to create the directory if it doesn't exist. The ```file.exists()``` check below ensures that the file will only be deleted if it actually exists.

Reply:
Fixed